src/labkey/Ajax.ts

```diff
@@ -306,6 +306,12 @@ export function request(config: RequestOptions): XMLHttpRequest {
 
             callback(success ? config.success : config.failure, scope, [xhr, config]);
             callback(config.callback, scope, [config, success, xhr]);
+        } else if (xhr.readyState === 2 && xhr.responseType === 'blob' ) {
+            // Change response type of download requests to allow reading any error messages upon failure
```

Comment:
Where is this expected to be read? I see that this path is not calling the "callbacks" provided so what actually shows the error to the user? Does the browser just throw up the text?

Reply:
this just changes the contenttype so the response can be read. It is still read when the readystate hits 4.

Reply:
Got it.